src/Controls/src/Core/Setter.cs

```diff
@@ -104,23 +100,89 @@ internal void UnApply(BindableObject target, bool fromStyle = false)
 			if (Property == null)
 				return;
 
-			object actual = targetObject.GetValue(Property);
-			if (!Equals(actual, Value) && Value is not BindingBase && Value is not DynamicResource)
-			{
+			targetObject.ClearValue(Property, specificity);
+		}
+	}
 
-				//Do not reset default value if the value has been changed
-				_originalValues.Remove(targetObject);
-				return;
-			}
+	//TODO: review this: merge FROM into a single int (vsm, manual, dynamicR, binding), and CSS into another
+	internal readonly struct SetterSpecificity : IComparable<SetterSpecificity>
```

Comment:
>SetterSpecificity [](http://example.com/codeflow?start=26&length=17)

This is is used through other part of the code. SetterSpecificity should probably be in his own file?